### DISCUSSION TITLE
[dynamic_link] Support for ofl parameter

### DISCUSSION THREAD

**Discussion by edisonpagtakhan** [firebase/flutterfire#5587] [Feature request]
**Is your feature request related to a problem? Please describe.**
Currently, there's no way to specify the `ofl` parameter in the dynamic link. This is helpful when the linked content does not have a web equivalent, thus we need to set a different URL for desktop users.

**Describe the solution you'd like**
Add a `desktopLink` for `DynamicLinkParameters`

**Additional context**
For reference: https://firebase.google.com/docs/dynamic-links/create-manually?authuser=0

**Comment by edisonpagtakhan** [comment]
A possible workaround for this is to manually create a long link, specify the `ofl`, then generate a short link using the plugin. However, it's harder to read compared to when using `DynamicLinkParameters`

**Comment by gkhristian** [comment]
I found this issue recently and I figure out a half-way workaround that can help to mitigate the issue. You create the dynamic link as it is specified and then modify the "queryParameters":

```
// WORKAROUND to add ofl parameter (fallback on desktop browser not yet implemented on FB DynamicLinks)
Uri uri = await parameters.buildUrl();
uri = uri.replace(queryParameters: Map<String, String>()..addAll(uri.queryParameters)..addAll({'ofl': YOUR_URI}));

final ShortDynamicLink dynamicUrl = await DynamicLinkParameters.shortenUrl(uri); // await parameters.buildShortLink();
```

**Reply by LazyBaneling** [reply]
What can we do now? shortenUrl has been removed in the new version.

**Comment by CalvinTp** [comment]
@gkhristian thanks the suggestion, it works while waiting for Flutter to support ofl parameter.

**Comment by guilherme-v** [comment]
😄 you can also try to create everything manually for now, it's not so complicated
```
    final url = Uri.parse('${your_subdomain..}/' +
        '?link=${..}' +
        '&apn=${...}  +
        '&ofl=https://${...}');

    final shortenedLink = await DynamicLinkParameters.shortenUrl(url);
    final shortUrl = shortenedLink.shortUrl;
```

**Reply by d3xt3r2909** [reply]
Unfortunately, this is not possible anymore from version 4.x.x

**Comment by myfknoll** [comment]
There is a new PR which allows adding a ofl parameter for short URLs
https://github.com/FirebaseExtended/flutterfire/pull/7796

**Comment by peer-f** [comment]
Do you have any information when It's going to be possible to configure the ofl parameter for dynamic links created in the Firebase console? There seems to be some demand for that [stackoverflow](https://stackoverflow.com/questions/73126009/firebase-console-dynamic-link-ofl-parameter). I want to create multiple deep links via the Firebase console that point to my apps on ios and android as well as my web app. Since the links need to be delivered via different mediums for example via email I want to store them in one place without creating new links in different places.
